### PR TITLE
FIX-#99: Upgrade min python version

### LINKED ISSUE #99: Upgrade min python version
pandas 1.4.0 requires python 3.8 and higher. We should upgrade min python version once https://github.com/modin-project/modin/pull/4036 is merged.

### PR DESCRIPTION
**Pull request by YarShev**
Signed-off-by: Igoshev, Yaroslav <yaroslav.igoshev@intel.com>

## What do these changes do?

- [x] passes `flake8 .`
- [x] passes `black .`
- [x] signed commit with `git commit -s` 
- [x] Resolves #99  
- [x] tests passing
- [x] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

**Comment by YarShev** [comment]
Guys, are you aware of the failure for MPI backend with python 3.9?

**Comment by no-ponomarev** [comment]
> Guys, are you aware of the failure for MPI backend with python 3.9?
As for me, no, something strange?

**Comment by prutskov** [comment]
> Guys, are you aware of the failure for MPI backend with python 3.9?

I've run only of 3.7/3.8

File: setup.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import pathlib
 from setuptools import setup, find_packages
-import sys
+
```
Comment (prutskov) [inline setup.py:3]:
```suggestion
```

Comment (YarShev) [reply setup.py:3]: I prefer to keep two empty lines between imports and code like in other python files.

**Comment by prutskov** [comment]
I propose to add python3.9 testing in a separate PR, because this is out of the issue scope and requires more efforts.

**Comment by YarShev** [comment]
@prutskov, that sounds reasonable to me.

**Comment by YarShev** [comment]
#104

**Comment by no-ponomarev** [comment]
LGTM, although some unrelated windows failure appeared

File: setup.py (hunk 1 above)
Comment (prutskov) [reply setup.py:3]: ok

**Comment by YarShev** [comment]
`test_get_ip` is a common failure that we have in GH actions. I think we can keep it mind.

**Review by prutskov — APPROVED** [review]
LGTM, assuming CI is green

**Comment by prutskov** [comment]
As decided, failures in `get_ip` test will be resolved later and it's not a stopper to merge (https://github.com/modin-project/unidist/pull/53#issuecomment-1023134544)

**Pull request merged**